### DISCUSSION TITLE
Joystick and Keyboard as one HID?

### DISCUSSION THREAD

**Discussion by s-celles** [NicoHood/HID#337] [Q&A] [answered]
Hello,

following discussion at https://forum.arduino.cc/t/joystick-and-keyboard-as-one-hid/458221 I'd like to know if I can use an Arduino Leonardo (only one board) as both joystick and keyboard.
My use case is about having a hat switch as part of a USB joystick and some buttons as part of a USB keyboard and some as USB joystick buttons

Kind regards

**Comment by NicoHood** [comment, marked as answer]
Yes. Use the gamepad of this library (just a different name) and add a keyboard. make sure to checkout the wiki to not exceed usb endpoints
https://github.com/NicoHood/HID/wiki/API-Documentation

**Reply by s-celles** [reply]
Thanks @NicoHood
